src/inline.jl

```diff
@@ -47,14 +48,29 @@ function limitstringmime(mime::MIME, x)
     return String(take!(buf))
 end
 
+const ipy_mime_json = [
+    "application/vnd.dataresource+json",
+    "application/vnd.vegalite.v2+json",
+    "application/vnd.vega.v3+json",
+    "application/vnd.vega.v4+json",
+]
+_display_dict(m::MIME, m_str, x) = Dict(m_str=>limitstringmime(m, x))
+# escape JSON string correctly before send_ipython
+for mime in ipy_mime_json
+    @eval begin
+        _display_dict(m::MIME{Symbol($mime)}, m_str, x) = Dict(m_str=>JSON.JSONText(limitstringmime(m, x)))
+        Base.Multimedia.istextmime(::MIME{Symbol($mime)}) = true
+    end
+end
+
 for mime in ipy_mime
     @eval begin
         function display(d::InlineDisplay, ::MIME{Symbol($mime)}, x)
             send_ipython(publish[],
                          msg_pub(execute_msg, "display_data",
                                  Dict(
                                   "metadata" => metadata(x), # optional
-                                  "data" => Dict($mime => limitstringmime(MIME($mime), x)))))
+                                  "data" => _display_dict(MIME($mime), $mime, x))))
```

Comment:
Rather than `MIME($mime)`, you should just give the argument `::MIME{Symbol($mime)}` a name, e.g. `m`, and use that.  This way `_display_dict` will be dispatched statically here.